workers/wfsfeature-harvester/src/main/java/org/fao/geonet/harvester/wfsfeatures/worker/EsWFSFeatureIndexer.java

```diff
@@ -173,346 +187,417 @@ public void deleteFeatures(Exchange exchange) {
 
-        // In WFS1.0.0, coordinates is lat/lon whatever the longitudeFirst is.
+        Query query = new Query();
         CoordinateReferenceSystem wgs84;
-
-        if (state.getParameters().getVersion().equals("1.0.0")) {
+        if (wfs.getInfo().getVersion().equals("1.0.0")) {
             wgs84 = CRS.getAuthorityFactory(true).createCoordinateReferenceSystem("EPSG:4326");
-//            Hints hints = new Hints(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER, Boolean.TRUE);
-//            CRSAuthorityFactory factory = ReferencingFactoryFinder.getCRSAuthorityFactory("EPSG", hints);
-//            wgs84 = factory.createCoordinateReferenceSystem("4326");
         } else {
-            wgs84 = CRS.getAuthorityFactory(true)
-                .createCoordinateReferenceSystem("urn:x-ogc:def:crs:EPSG::4326");
+            wgs84 = CRS.getAuthorityFactory(true).createCoordinateReferenceSystem("urn:x-ogc:def:crs:EPSG::4326");
         }
+        query.setCoordinateSystemReproject(wgs84);
 
         try {
-            FeatureSource<SimpleFeatureType, SimpleFeature> source = wfs.getFeatureSource(typeName);
-            Extent crsExtent = wgs84.getDomainOfValidity();
-            ReferencedEnvelope wgs84bbox = null;
-            for (GeographicExtent element : crsExtent.getGeographicElements()) {
-                if (element instanceof GeographicBoundingBox) {
-                    GeographicBoundingBox bounds = (GeographicBoundingBox) element;
-                    wgs84bbox = new ReferencedEnvelope(
-                            bounds.getSouthBoundLatitude(),
-                            bounds.getNorthBoundLatitude(),
-                            bounds.getWestBoundLongitude(),
-                            bounds.getEastBoundLongitude(),
-                            wgs84
-                    );
-                }
-            }
-
-
-            // TODO: GeoServer WFS 1.0.0 in some case return
-//            Feb 18, 2016 12:04:22 PM org.geotools.data.wfs.v1_0_0.NonStrictWFSStrategy createFeatureReaderGET
-//            WARNING: java.io.IOException: org.xml.sax.SAXException: cannot merge two target namespaces. http://www.openplans.org/topp http://www.openplans.org/spearfish
-            // Retrieve features in WGS 84
-            Query query = new Query();
-            query.setCoordinateSystemReproject(wgs84);
-            FeatureCollection<SimpleFeatureType, SimpleFeature> featuresCollection =
-                source.getFeatures(query);
-
-            final FeatureIterator<SimpleFeature> features = featuresCollection.features();
-            int numInBatch = 0, nbOfFeatures = 0;
-            Map<String, String> docCollection = new HashMap<>();
-
-            saveHarvesterReport(state);
-
-            String titleExpression = state.getParameters().getTitleExpression();
-            String defaultTitleAttribute = titleExpression == null ?
-                WFSFeatureUtils.guessFeatureTitleAttribute(featureAttributes) : null;
+            nbOfFeatures = 0;
```

Comment:
Seems that many indexations can be triggered the same time, maybe have nbOfFeature at method scope should be better ? Need clarification in my opinion.

Reply:
WFSFeatureIndexer is not a singleton so it can't run indexFeatures multiple time.  It has been fixed in https://github.com/camptocamp/sextant-geonetwork/commit/c769ca1db4e7f3838ef603f038ba044f898f05fa and is reported here in this PR. So it should be ok no ?

Reply:
Yes, it should. Thank you.